app/models/punch.rb

```diff
@@ -5,6 +5,7 @@
 #  id         :integer          not null, primary key
 #  in         :boolean          not null
 #  user_id    :integer          not null
+#  reason     :string
```

Comment:
Nice 👌 

If you want to ensure only a set of values can ever be set here, add [an inclusion validation](http://guides.rubyonrails.org/active_record_validations.html#inclusion).

Something like `validates :size, inclusion: { in: %w[personal_project class_project event class_session other] }`